pkl-core/src/test/files/LanguageSnippetTests/input/listings/typeCheck.pkl

```diff
@@ -1,41 +1,367 @@
 import "pkl:test"
```

Comment:
Note: these don't amend `pkl:test` because `facts` and `examples` are declared as `Mapping<String, Listing<Boolean>>` and `Mapping<String, Listing<unknown>>`. These are typed properties that themselves use Mapping/Listing type checks, and problems with typechecking can quickly be conflated between the tests themselves and the test framework.